ext/oauth-client/Civi/Api4/OAuthSessionToken.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Civi\Api4;
+
+use Civi\Api4\Generic\AbstractAction;
+use Civi\Api4\Generic\Result;
+
+class OAuthSessionToken extends Generic\AbstractEntity {
+
+  const ENTITY = 'OAuthSessionToken';
+
+  public static function create($checkPermissions = TRUE): Generic\BasicCreateAction {
+    $action = new Generic\BasicCreateAction(
+      self::ENTITY,
+      __FUNCTION__,
+      function ($item, $createAction) {
+        $session = \CRM_Core_Session::singleton();
+        $all = $session->get('OAuthSessionTokens') ?? [];
+        $all[] = &$item;
+        $item['cardinal'] = array_key_last($all);
```

Comment:
Copied from [my response in chat](https://chat.civicrm.org/civicrm/pl/ejdse9qp4i8ufjtar31r4ejx5o):

@totten as you guessed, I wanted to choose a name other than `id` because this number is unique only in the limited context of the session. But I did feel like it would be good to have a field that could be used to reliably address a single `OAuthSessionToken`, since none of the other fields can be relied on for that purpose.

As to it being stable, well... that's something I want to work on. I implemented the `deleteAll` action in the simplest way that came to mind, but after playing around with it I see that it resets the `cardinal` counter, meaning a different token could be created with the same `cardinal` later in the same session. I have a hard time foreseeing a case where that would be a problem, but it should be easy enough to fix.